src/Job/JobTrait.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace AgDevelop\ForkingSupervisor\Job;
+
+use AgDevelop\ForkingSupervisor\Watchdog\Watchdog;
+use Psr\Log\LoggerInterface;
+
+trait JobTrait
+{
+    private Watchdog $watchdog;
+    private LoggerInterface $logger;
+    private int $retries=0;
+    private bool $shouldRetryOnFail = true;
+    private readonly string $jobId;
+
+    public function setShouldRetryOnFail(bool $shouldRetryOnFail): self {
+        $this->shouldRetryOnFail = $shouldRetryOnFail;
+        return $this;
+    }
+
+    public function shouldRetryOnFail(): bool
+    {
+        return $this->shouldRetryOnFail;
+    }
+
+    public function incrementRetries(): void {
+        $this->retries++;
+    }
+
+    public function setJobId(string $id): self
+    {
+        $this->jobId = $id;
+
+        return $this;
+    }
+
+    public function setLogger(LoggerInterface $logger): self
+    {
+        $this->logger = $logger;
+
+        return $this;
+    }
+
+    public function setWatchdog(Watchdog $watchdog): self
+    {
+        $this->watchdog = $watchdog;
+        return $this;
```

Comment:
O tu nie ma entera, a w 33 i 40 jest. To celowe działanie? :D

Reply:
no nie celowe, jak ustalamy ma być czy nie ma być?

Reply:
jw